dallinger/models.py

```diff
@@ -1373,6 +1373,11 @@ class Vector(Base, SharedMixin):
 
     __tablename__ = "vector"
 
+    #: A String giving the name of the class. Defaults to
+    #: ``vector``. This allows subclassing.
+    type = Column(String(50), default="vector", server_default="vector")
```

Comment:
I'm wondering about the `default` and `server_default` values here, since these have not been used in the other subclasses of `Base`. My mind immediately goes to "what's the difference? what should I copy if I'm making my own model, or if I'm subclassing one of the Dallinger built-in models?" `Node` is frequently subclassed and it also has a `type` column. Should we apply similar defaults to `Node` and others? (or would this cause harm?)

Reply:
We only need it here to maintain back compatibility when importing old datasets that didn't have this column. We could apply it to the other models for consistency but I don't think it would have any meaningful function. Would you prefer me to do that?

Reply:
I forgot that you're _adding_ the `type` column. Now I get it. I'd propose just adding a comment, since that will both explain why the defaults are there, and be a possible clue to the next person who needs to do something similar:

The `type` column was added in 9/2022, 7+ years after the `Vector` ORM class was introduced. To support
datasets which don't include this column we define a default value which will be used when object instances are loaded.

(or something along these lines... I'm not sure that's exactly correct!)

Reply:
Thanks @jessesnyder, good suggestion, have implemented it!